overlord/hookstate/ctlcmd/get.go

```diff
@@ -22,23 +22,32 @@ package ctlcmd
 import (
 	"encoding/json"
 	"fmt"
+	"strings"
 
 	"github.com/snapcore/snapd/i18n/dumb"
 	"github.com/snapcore/snapd/overlord/configstate"
+	"github.com/snapcore/snapd/overlord/hookstate"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/snap"
 )
 
 type getCommand struct {
 	baseCommand
 
+	// these two options are mutually exclusive
+	ForceSlotSide bool `long:"slot" description:"request attribute of the slot"`
+	ForcePlugSide bool `long:"plug" description:"request attribute of the plug"`
+
 	Positional struct {
-		Keys []string `positional-arg-name:"<keys>" description:"option keys" required:"yes"`
-	} `positional-args:"yes" required:"yes"`
+		PlugOrSlotSpec string   `positional-args:"true" positional-arg-name:":<plug|slot>" required:"yes"`
```

Comment:
You're right, that's exactly what happened. I've removed 'required' here and added an explicit case in Execute, plus a test case that covers this scenario for both get and set.